### PR TITLE
Implement JWT based auth (WIP)

### PR DESCRIPTION
**Pull request by brocaar**
See #20

### CODE REVIEW

File: internal/api/auth/auth.go [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/brocaar/lorawan"
+	jwt "github.com/dgrijalva/jwt-go"
+)
+
+// Claims defines the struct containing the token claims.
+type Claims struct {
```
Comment (ivajloip) [inline internal/api/auth/auth.go]: Did you consider using [`StandardClaims`][1]? I see why you might want to implement your own solution, but I think that embedding this structure could also provide some benefits.

[1]: https://github.com/dgrijalva/jwt-go/blob/d2709f9f1f31ebcda9651b03077758c1f3a0018c/claims.go#L18

**Comment by brocaar** [comment]
Do you mean as a replacement for the current `Claims` structure, or embedding the `StandardClaims` into `Claims`?

**Comment by ivajloip** [comment]
Embedding `StandardClaims` into `Claims`. It has `exp` and other fields validations built-it, which could be handy, but not too difficult to implement yourself if you want them.

**Comment by brocaar** [comment]
Good one, will do!

File: internal/api/auth/auth.go (hunk 1 above)
Comment (brocaar) [inline internal/api/auth/auth.go]: Here you go: https://github.com/brocaar/loraserver/pull/53/commits/381711e58a7194efcfeb5b0b6ba5174943a4d147

File: internal/api/auth/auth.go [hunk 2]
```diff
@@ -39,22 +37,9 @@ type Claims struct {
...
-
-// Valid returns if the claim is valid (it hasn't expired yet).
-// It doesn't validate the content of the claim (
+// Valid validates the standard claims of the token.
 func (c Claims) Valid() error {
-	if !c.validateExpiration(false) {
-		return errors.New("token has expired")
-	}
-	return nil
+	return c.StandardClaims.Valid()
```
Comment (ivajloip) [inline internal/api/auth/auth.go]: I would expect that you can directly *inherit* this method from `StandardClaims`. Maybe you consider expanding it and that's why you decided to provide it on your own?

Comment (brocaar) [inline internal/api/auth/auth.go]: Whoops.. you're right :-)

**Pull request merged**